### ISSUE TITLE
Outputs from GRU layer

### ISSUE THREAD

**Issue by peastman** [deepchem/deepchem#570]
I'm trying to use the GRU layer to create a recurrent policy network for RL.  I'm confused by certain aspects of it.

The layer has separate `n_hidden` and `out_channels` properties, which confused me since a GRU doesn't have separate outputs that are distinct from its internal state.  But looking at the implementation, it adds a linear layer (which is hardcoded to use a sigmoid activation function) to its output, and the `out_channels` argument sets the size of that layer.  What's the reason for that?  If I want to add a linear layer after the GRU I can do that as a separate layer.  But I may not always want that, and even when I do want it, I won't always want a sigmoid activation function.

The code for it also confuses me:

```Python
projection = lambda x: tf.contrib.layers.linear(x, num_outputs=self.out_channels, activation_fn=tf.nn.sigmoid)
out_tensor = tf.map_fn(projection, rnn_outputs)
```

But `rnn_outputs` is a single tensor, not a list of tensors, so why the need for the `tf.map_fn()`?  Also, the Tensorflow documentation doesn't mention any such function as `tf.contrib.layers.linear()`.  But after a bit of searching online, I found the definition of it in the source code: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/layers.py#L2242.  It's an alias to `fully_connected` that hardcodes the activation function to `None`.  Which makes me even more confused about what's going on here!

**Comment by lilleswing** [comment]
@peastman we should split the layer up.  No one is using it (except maybe @DorisMai ).  I made if for a pure tf port of https://github.com/maxhodak/keras-molecules when I didn't really know what I was doing. 

the tf.map_fn() is to do a dimensionality reduction on a per character basis to the number of characters in the input language.

@DorisMai are you using this layer?  or are you still on the pure keras implementation?

**Comment by peastman** [comment]
Ok, I'll remove the extra layer.  If someone wants it, they can just add a Dense layer following it.

**Comment by patrickhop** [comment]
If you guys are moving into RL stuff, I have an implementation of [https://arxiv.org/abs/1511.05952] that I'd be happy to share.

**Comment by lilleswing** [comment]
@patrickhop yeah just throw up a CR to put it in contrib/rl for the experience replay.

**Comment by rbharath** [comment]
Going to close since I think this issue has been resolved.
